### ISSUE TITLE
Product Gallery: Displays error after page refresh in Editor

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#52936]
**Steps to reproduce:**

1. Go to Single Product template
2. Replace Product Image Gallery with Product Gallery (Beta)
3. Save the template and refresh the page

**Expected:** Block displays normally
**Actual:** There's error displayed:

```
**Sorry, an error occurred**
The following error was returned
productGalleryClientId and clientId codes mismatch.
```

**Video:**

https://github.com/user-attachments/assets/3e555267-fffb-41d9-94b4-9f99fcb8f1e8

**Comment by wpchannel** [comment]
Same error here even after restoring a revision.

**Comment by rehayema** [comment]
Looks like the issue has returned to the Product Gallery (Beta) block. I saw it on a user's site -9287357-zd-a8c

**Comment by kmanijak** [comment]
Thanks for ping @rehayema! The fix landed in WooCommerce 9.6 which is due to release on January 20, 2025 so not yet published 🙌
